lib/control-connection.js

```diff
@@ -141,7 +245,7 @@ ControlConnection.prototype.getConnection = function (callback) {
  * it's the first time that trying to retrieve host information
  * @param {Function} callback
  */
-ControlConnection.prototype.initOnConnection = function (firstTime, callback) {
+ControlConnection.prototype.refreshOnConnection = function (firstTime, callback) {
   var c = this.connection;
   var self = this;
   self.log('info', 'Connection acquired, refreshing nodes list');
```

Comment:
I noticed that subscribeHostEvents subscribes to 'down' events on a host, but the listener is never removed when the host goes down.  Is there something that prevents receiving down events for an old host?  I think as it would work right now, the hostDownHandler could get called for a host that was previously the control connection going down.  It seems like in this case we would detect that a connection is still set and log that the control connection is down and it's attempting to get one, but it wouldn't actually replace the connection (fortunately).   I'll test this out.